File: client/src/api/games.ts
import { apiClient } from '../lib/api';

export interface GameEntry {
  app_id: string;
  name: string;
  size_bytes: number;
}

export interface GameLibrary {
  provider: string;
  provider_name: string;
  path: string;
  device_id: number | null;
  total_bytes: number;
  game_count: number;
  games: GameEntry[];
}

export interface GameLibrariesResponse {
  libraries: GameLibrary[];
  total_bytes: number;
  available: boolean;
}

export const getGameLibraries = async (): Promise<GameLibrariesResponse> => {
  const response = await apiClient.get('/api/games/libraries');
  return response.data;
};

File: backend/app/schemas/games.py
"""Schemas for game library storage usage."""
from typing import List, Optional

from pydantic import BaseModel


class GameEntry(BaseModel):
    """A single installed game/app within a library."""
    app_id: str
    name: str
    size_bytes: int


class GameLibrary(BaseModel):
    """One game library (a directory holding installed games)."""
    provider: str          # e.g. "steam"
    provider_name: str     # e.g. "Steam"
    path: str              # absolute library path on disk
    device_id: Optional[int] = None  # os.stat().st_dev, for mountpoint matching
    total_bytes: int
    game_count: int
    games: List[GameEntry]  # sorted by size_bytes descending


class GameLibrariesResponse(BaseModel):
    """All discovered game libraries across providers."""
    libraries: List[GameLibrary]
    total_bytes: int        # sum across all libraries
    available: bool         # True if at least one provider is available
